project/index.mjs

```diff
@@ -0,0 +1,173 @@
+    ],
+    exposedHeaders: '*',
+    origin: '*',
+    maxAge: '600',
+  })
+)
+
+// Send a successful response with the appropriate JSON or alternate response based on params
+export function respondWithProject(req, res, project) {
+  const id = project['@id'] ?? project.id ?? null
+
+  let textType = req.query.text
+  let image = req.query.image
+  let lookup = req.query.lookup
+  let view = req.query.view
+  let responseType = [textType, image, lookup, view]
+    .find(elem => elem !== undefined) // Only one response type is valid
+
+  let embed = req.query.embed
+
+  if (!responseType || responseType === view && view == "json") {
+    res.set('Content-Type', 'application/json; charset=utf-8')
+    res.location(id)
+    res.status(200)
+    res.json(project)
+    return
+  }
+
+  let retVal;
+  switch (responseType) {
+
+    case textType:
+      switch (textType) {
+        case "blob":
+          res.set('Content-Type', 'text/plain; charset=utf-8')
+          // return: a complete blob of text of all lines concatenated
+          /* retVal = new Blob(
+            project.layers.map(layer =>
+              db.getByID(layer).getLines().map(line => line.textualBody).join(" ")
+            ).join(" ")
+          ) */
+          retVal = new Blob('mock text')
+          break
+        case "layers":
+          res.set('Content-Type', 'application/json; charset=utf-8')
+          /* retVal = project.layers.map(layer => db.getByID(layer)) */
+          retVal = [['mock text'], ['mock text']]
+          break
+        case "pages":
+          res.set('Content-Type', 'application/json; charset=utf-8')
+          /* retVal = project.layers.flatMap(layer => db.getByID(layer).getPages()) */
+          retVal = [['mock text'], ['mock text']]
+          break
+        case "lines":
+          res.set('Content-Type', 'application/json; charset=utf-8')
+          retVal = [['mock text'], ['mock text']]
+          break
+        default:
+          utils.respondWithError(res, 400, 
+            'Improper request.  Parameter "text" must be "blob," "layers," "pages," or "lines."')
+          break
+      }
+      break
+
+    case image:
+      switch (image) {
+        case "thumb":
+          res.set('Content-Type', 'text/uri-list; charset=utf-8')
+          // return: the URL of the default resolution of a thumbnail from the Manifest
+          retVal = 'https://example.com'
+          // make sure to handle this differently if req.query.embed is true
+          break
+        default:
+          utils.respondWithError(res, 400, 
+            'Improper request.  Parameter "image" must be "thumbnail."')
+          break
+      }
+      break
+
+    case lookup:
+      switch (lookup) {
+        case "manifest":
+          // return: (layers[], annotations[], metadata[], group) find the related document or Array of documents and return that instead, the version allowed without authentication
+          retVal = [['mock text'], ['mock text']]
```

Comment:
This will be a resolved JSON manifest.  The value for `manifest` in the return will not be the URI, it will be the actual JSON of the resolved URI.